dali/operators/decoder/nvjpeg/nvjpeg_decoder_decoupled_api.h

```diff
@@ -685,18 +691,25 @@ class nvJPEGDecoder : public Operator<MixedBackend>, CachedDecoderImpl {
       memset(nvjpeg_destinations_.data(), 0, nvjpeg_destinations_.size() * sizeof(nvjpegImage_t));
 
       int j = 0;
+      TensorVector<CPUBackend> tv(samples_hw_batched_.size());
+
       for (auto *sample : samples_hw_batched_) {
         assert(!sample->roi);
         int i = sample->sample_idx;
         const auto &in = ws.Input<CPUBackend>(0, i);
         const auto &out_shape = output_shape_.tensor_shape(i);
 
-        in_data_[j] = in.data<uint8_t>();
+        tv[j].ShareData(const_cast<Tensor<CPUBackend>*>(&in));
         in_lengths_[j] = in.size();
         nvjpeg_destinations_[j].channel[0] = output.mutable_tensor<uint8_t>(i);
         nvjpeg_destinations_[j].pitch[0] = out_shape[1] * out_shape[2];
         j++;
       }
+
+      hw_decoder_images_staging_.Copy(tv, 0);
+      for (size_t k = 0; k < samples_hw_batched_.size(); ++k) {
+        in_data_[k] = hw_decoder_images_staging_.mutable_tensor<uint8_t>(k);
+      }
       CUDA_CALL(cudaEventSynchronize(hw_decode_event_));
```

Comment:
What exactly is this syncrhonization here for? Doesn't nvjpegDecodeBatched have well defined stream order?

Reply:
Yes, it has. It launches cudaMemcpyAsyn and kernels for color conversion.